modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.txt

```diff
@@ -165,3 +165,30 @@ class org.elasticsearch.search.lookup.FieldLookup -> org.elasticsearch.search.lo
   List getValues()
   boolean isEmpty()
 }
+
+class org.elasticsearch.index.similarity.ScriptedSimilarity$Query -> org.elasticsearch.index.similarity.ScriptedSimilarity$Query extends Object {
```

Comment:
For the whitelisted classes you can change the type name to be like the following:

```class org.elasticsearch.index.similarity.ScriptedSimilarity.Query -> org.elasticsearch.index.similarity.ScriptedSimilarity$Query extends Object {```

Note the '$' got changed to '.' in the Painless name of the type (first piece) while it still represents the appropriate Java class (second piece).  All the rest of the defined types will work this way too.